src/hotspot/share/runtime/arguments.cpp

```diff
@@ -1110,7 +1110,16 @@ void Arguments::print_on(outputStream* st) {
   st->print_cr("java_command: %s", java_command() ? java_command() : "<unknown>");
   if (_java_class_path != NULL) {
     char* path = _java_class_path->value();
-    st->print_cr("java_class_path (initial): %s", strlen(path) == 0 ? "<not set>" : path );
+    size_t len = strlen(path);
+    st->print("java_class_path (initial): ");
+    if (len < O_BUFLEN) {
+      st->print_cr("%s", len == 0 ? "<not set>" : path );
+    } else {
+      for (size_t i = 0; i < len; i++) {
+        st->put(path[i]);
+      }
+      st->cr();
+    }
```

Comment:
Maybe we can use `st->print_raw()` instead?

Reply:
Thanks for your review.
I've changed it to st->print_raw_cr() so that st->cr() is not needed.
I also removed the "@requires vm.cds" from the test since it is unrelated to cds.

Reply:
How about doing this? This will make sure the print_raw_cr() path is tested (i.e., when generating hs_err files).

```
  if (len == 0) {
    st->print_raw_cr("<not set>");
  } else {
    st->print_raw_cr(path, len);
  }
```

Reply:
Please add a comment why print_raw is used.